sled-agent/src/sim/disk.rs

```diff
@@ -8,32 +8,211 @@ use crate::nexus::NexusClient;
 use crate::params::DiskStateRequested;
 use crate::sim::simulatable::Simulatable;
 use async_trait::async_trait;
+use dropshot::ConfigDropshot;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
 use omicron_common::api::external::DiskState;
 use omicron_common::api::external::Error;
 use omicron_common::api::external::Generation;
 use omicron_common::api::internal::nexus::DiskRuntimeState;
+use omicron_common::api::internal::nexus::ProducerEndpoint;
+use oximeter_producer::Server as ProducerServer;
 use propolis_client::api::DiskAttachmentState as PropolisDiskState;
+use std::net::{Ipv6Addr, SocketAddr};
 use std::sync::Arc;
+use std::time::Duration;
 use uuid::Uuid;
 
 use crate::common::disk::{Action as DiskAction, DiskStates};
 
+// Oximeter timeseries names are derived based on the precise names of structs,
+// so we shove this in a module to more liberally use arbitrary names (like
+// "Read").
+mod producers {
+    use super::*;
+    use oximeter::{
+        types::{Cumulative, Sample},
+        Metric, Target,
+    };
+
+    #[derive(Debug, Clone, Target)]
+    pub struct CrucibleUpstairs {
+        pub upstairs_uuid: Uuid,
+    }
+
+    // TODO: It would be a lot nicer if we could just depend on the Crucible
```

Comment:
Yeah, this seems like a bummer. The `crucible` repo depends on `oximeter` from the `main` branch of `omicron`. I don't think there's been an update to `oximeter` since that dependency was put in, so I'm not sure if we'll _always_ have this problem or not.